pkg/master/master.go

```diff
@@ -193,33 +194,39 @@ func (m *Master) InstallAPIs(c *Config) {
 		m.thirdPartyStorage = c.StorageDestinations.APIGroups[extensions.GroupName].Default
 		m.thirdPartyResources = map[string]thirdPartyEntry{}
 
-		expVersion := m.experimental(c)
-
-		if err := expVersion.InstallREST(m.HandlerContainer); err != nil {
-			glog.Fatalf("Unable to setup experimental api: %v", err)
+		extensionResources := m.getExtensionResources(c)
+		extensionsGroupMeta := latest.GroupOrDie(extensions.GroupName)
+		// Update the prefered version as per StorageVersions in the config.
+		storageVersion, found := c.StorageVersions[extensionsGroupMeta.GroupVersion.Group]
+		if !found {
+			glog.Fatalf("Couldn't find storage version of group %v", extensionsGroupMeta.GroupVersion.Group)
 		}
-		g, err := latest.Group(extensions.GroupName)
+		preferedGroupVersion, err := unversioned.ParseGroupVersion(storageVersion)
 		if err != nil {
-			glog.Fatalf("Unable to setup experimental api: %v", err)
+			glog.Fatalf("Error in parsing group version %s: %v", storageVersion, err)
 		}
-		expAPIVersions := []unversioned.GroupVersionForDiscovery{
-			{
-				GroupVersion: expVersion.GroupVersion.String(),
-				Version:      expVersion.GroupVersion.Version,
-			},
+		extensionsGroupMeta.GroupVersion = preferedGroupVersion
```

Comment:
This is correct as discussed.
Added a for loop in genericserver to install all versions in the group.